### ISSUE TITLE
Example (upload a new file) needs C# example

### ISSUE THREAD

**Issue by zipswich** [microsoftgraph/microsoft-graph-docs#6815]
It is difficult to derive the C# code from the HTTP example. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d92fb8c7-4409-d945-57dd-5d6fc4010051
* Version Independent ID: f77c05bd-2688-bd4e-1f8f-f18feb3a8639
* Content: [Upload small files - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/driveitem-put-content?view=graph-rest-1.0&tabs=csharp)
* Content Source: [api-reference/v1.0/api/driveitem-put-content.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/v1.0/api/driveitem-put-content.md)
* Product: **sharepoint**
* Technology: **microsoft-graph**
* GitHub Login: @JeremyKelley
* Microsoft Alias: **JeremyKelley**

**Comment by michael-hawker** [comment]
In addition the update example uses a string type for the stream, which isn't the right-type.  Showing loading a file stream or converting the text to the stream as a complete example would be more useful.  

Also related to #3902 about setting the proper file type for upload should be part of this sample.

**Comment by angelinatquach** [comment]
Please add a c#/non-http example for uploading a NEW file with microsoft graph. Not sure why it's been a month and still no documentation is provided

**Comment by zipswich** [comment]
@angelinatquach in case it is useful and you have not noticed it, here is a working code snippet for both small and large files:
https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3097#issuecomment-577906540

**Comment by angelinatquach** [comment]
@zipswich  your example only seems to reference updating an existing file, since your snippet always requires an item ID
`.Items[diameraAlternativeRoll.Id]`

I want to upload a NEW file, so I wouldn't know the item id

**Comment by zipswich** [comment]
@angelinatquach diameraAlternativeRoll.Id is the id of an **existing** folder where a file will be uploaded.

**Comment by angelinatquach** [comment]
@zipswich  is the existing folder the same thing as a drive? 

This is the error I get. Neither my folderId string nor my fileName string have the / in there, so not sure where the extra slash comes from of if it matters

Microsoft.Graph.ServiceException: Code: -1, Microsoft.SharePoint.Client.InvalidClientQueryException
Message: The expression "drive/items/a!SEZda52sE14a-WDxCe6A1DBhDaNZOGnFSQv-ebsW5b__vKAScHBQDFNwrjbS0N2a//Document.docx/uploadContent" is not valid.

**Comment by zipswich** [comment]
@angelinatquach No. It is a folder under OneDrive.  Here is the code to get it:
```
     DriveItem diameraAlternativeRoll = null;
     IDriveItemChildrenCollectionPage children = await provider.Graph.Me.Drive.Root.Children
                                    .Request()
                                    .GetAsync();
     diameraAlternativeRoll = children.First(r => r.Name == "Camera Alternative roll");
```

**Comment by angelinatquach** [comment]
thank you!!! This worked so good, I really appreciate it @zipswich

**Comment by michael-hawker** [comment]
@lramosvea why was the `needs more info` tag added? I think the ask is pretty clear. There's a C# example for updating an existing sample on the page, but no example for adding a new file.

**Comment by lramosvea** [comment]
Apologies @michael-hawker, didn't mean to add that label. I am reopening this issue, so that the corresponding team can work on it. Thanks!
